app/controllers/admin/document_searches_controller.rb

```diff
@@ -1,6 +1,54 @@
 class Admin::DocumentSearchesController < Admin::BaseController
   def show
-    filter_options = params.slice(:title, :type, :subtypes, :state).reverse_merge(state: 'active', per_page: 10)
-    @filter = Admin::EditionFilter.new(Edition, current_user, filter_options)
+    @editions = Filterer.new(params).editions
+  end
+
+  class Filterer
+    attr_reader :params
+
+    def initialize(params)
+      @params = params
+    end
+
+    def editions
+      filters_for(params).inject(edition_scope) do |editions, filter|
+        filter.call(editions)
+      end
+    end
+
+    def edition_scope
+      Edition
+        .with_translations(I18n.locale)
+        .limit(10)
+    end
+  private
+    def filters_for(params)
+      filters = []
+
+      if params[:title]
+        filters << ->(editions) { editions.with_title_containing(params[:title]) }
+      end
+
+      type = find_type(params[:type])
+      if type
+        filters << ->(editions) { editions.by_type(type) }
+
+        if params[:subtypes]
+          filters << ->(editions) { editions.by_subtypes(type, params[:subtypes]) }
+        end
+      end
+
+      filters << ->(editions) { editions.public_send(params[:state] || 'active' ) }
```

Comment:
How safe is this? What if a search was submitted with `params[:state] = "destroy_all"`?